sonic_db_config/db_config.go

```diff
@@ -194,6 +195,175 @@ func GetDbTcpAddr(db_name string, ns string) (addr string, err error) {
 	return hostname + ":" + strconv.Itoa(port), err
 }
 
+func GetDbDefaultInstance() (dbkey swsscommon.SonicDBKey, err error) {
+	dbkey = swsscommon.NewSonicDBKey()
+	dbkey.SetNetns(SONIC_DEFAULT_NAMESPACE)
+	dbkey.SetContainerName(SONIC_DEFAULT_CONTAINER)
+	return dbkey, nil
+}
+
+func CheckDbMultiInstance() (ret bool, err error) {
+	if !sonic_db_init {
+		err = DbInit()
+		if err != nil {
+			return false, err
+		}
+	}
+	defer CatchException(&err)
+	dbkey_vec := swsscommon.SonicDBConfigGetDbKeys()
+	length := int(dbkey_vec.Size())
+	// If there are more than one instances, this means that SONiC is using multi-namespace or multi-container
+	return length > 1, err
+}
+
+func GetDbNonDefaultInstances() (dbkey_list []swsscommon.SonicDBKey, err error) {
+	if !sonic_db_init {
+		err = DbInit()
+		if err != nil {
+			return dbkey_list, err
+		}
+	}
+	defer CatchException(&err)
+	dbkey_vec := swsscommon.SonicDBConfigGetDbKeys()
+	// Translate from vector to array
+	length := int(dbkey_vec.Size())
+	for i := 0; i < length; i += 1 {
+		dbkey := dbkey_vec.Get(i)
+		if dbkey.GetNetns() == SONIC_DEFAULT_NAMESPACE && dbkey.GetContainerName() == SONIC_DEFAULT_CONTAINER {
+			continue
+		}
+		dbkey_list = append(dbkey_list, dbkey)
+	}
+	return dbkey_list, err
+}
+
+func GetDbAllInstances() (dbkey_list []swsscommon.SonicDBKey, err error) {
+	if !sonic_db_init {
+		err = DbInit()
+		if err != nil {
+			return dbkey_list, err
+		}
+	}
+	defer CatchException(&err)
+	dbkey_vec := swsscommon.SonicDBConfigGetDbKeys()
+	// Translate from vector to array
+	length := int(dbkey_vec.Size())
+	for i := 0; i < length; i += 1 {
+		dbkey := dbkey_vec.Get(i)
+		dbkey_list = append(dbkey_list, dbkey)
+	}
+	return dbkey_list, err
+}
+
+func GetDbInstanceFromTarget(ns string, container string) (dbkey swsscommon.SonicDBKey, ret bool) {
+	dbkey_list, err := GetDbAllInstances()
+	if err != nil {
+		return nil, false
+	}
+	for _, dbkey := range dbkey_list {
```

Comment:
>for [](http://example.com/codeflow?start=1&length=3)

O(n) time complexity. Possible to improve?

Reply:
Fixed